datadog_checks_base/datadog_checks/base/log.py

```diff
@@ -66,6 +67,28 @@ def emit(self, record):
         datadog_agent.log(msg, record.levelno)
 
 
+class SanitizationFilter(logging.Filter):
+    """
+    A filter for sanitizing log records messages.
+    """
+
+    def __init__(self, name, sanitize):
+        # type: (str, Callable[[str], str]) -> None
+        super(SanitizationFilter, self).__init__(name)
+        self.sanitize = sanitize
+
+    def filter(self, record):
+        # type: (logging.LogRecord) -> bool
+        record.msg = self.sanitize(to_native_string(record.msg))
+
+        if isinstance(record.args, dict):
+            record.args = {key: self.sanitize(value) for key, value in iteritems(record.args)}
+        else:
+            record.args = tuple(self.sanitize(arg) for arg in record.args)
```

Comment:
Is is possible to avoid this by not hooking in until later when all we have is a string?

Reply:
It's possible using a `Formatter`. We'd need to refactor the current logging setup to extract an `AgentLogFormatter` (currently formatting is done inside the `AgentLogHandler`), and then wrap it around a `SanitizationFormatter`.

I pushed this in 397039ad6, it might be worth extracting the refactor as a separate PR, WDYT?